src/fluent-bit.c

```diff
@@ -550,25 +548,6 @@ static void flb_signal_exit(int signal)
         flb_print_signal(SIGTERM);
         flb_print_signal(SIGSEGV);
     };
-
-    /* Signal handlers */
-    /* SIGSEGV is not handled here to preserve stacktrace */
-    switch (signal) {
-    case SIGINT:
-    case SIGTERM:
-#ifndef FLB_SYSTEM_WINDOWS
-    case SIGQUIT:
-    case SIGHUP:
-#endif
-        if (cf_opts != NULL) {
-            flb_cf_destroy(cf_opts);
-        }
-        flb_stop(ctx);
-        flb_destroy(ctx);
```

Comment:
These functions will be called 
https://github.com/fluent/fluent-bit/blob/824ba3dd084e2a72a2f2e2a833d5d44d46f61341/src/fluent-bit.c#L1348-L1352
and
https://github.com/fluent/fluent-bit/blob/824ba3dd084e2a72a2f2e2a833d5d44d46f61341/src/fluent-bit.c#L1364-L1365